core/amber/src/main/python/core/architecture/managers/operator_manager.py

```diff
@@ -1,8 +1,127 @@
-from typing import Optional
+import inspect
+import sys
+import uuid
+import fs
+import importlib
+from loguru import logger
+from pathlib import Path
+from typing import Tuple, Optional, Mapping
 
 from core.models import Operator
 
 
+def gen_uuid(prefix=""):
+    return f"{prefix}{'-' if prefix else ''}{uuid.uuid4().hex}"
+
+
 class OperatorManager:
     def __init__(self):
+        # create a tmp fs for storing source code, which will be removed when the
+        # workflow is completed.
+        # TODO:
+        #       For various reasons when the workflow is not completed successfully,
+        #  the tmp fs could not be closed properly. This means it may leave files
+        #  in the /var/tmp folder after a partially started or failed execution.
+        #       A full-life-cycle management of tmp fs is required to consider all
+        #  possible errors happened during execution. However, the full-life-cycle
+        #  management could be hard due to errors from JAVA side which causes force
+        #  kill on the Python process.
+        #       As each python file is usually tiny in size, and the OS can
+        #  periodically clean up /var/tmp anyway, the full-life-cycle management is
+        #  not a priority to be fixed.
+        self.fs = fs.open_fs("temp://")
+        self.root = Path(self.fs.getsyspath("/"))
+        logger.info(f"Opening a tmp directory at {self.root}.")
+        sys.path.append(str(self.root))
         self.operator: Optional[Operator] = None
+
+    @staticmethod
+    def gen_module_file_name() -> Tuple[str, str]:
+        """
+        Generate a UUID to be used as udf source code file.
+        :return Tuple[str, str]: the pair of module_name and file_name.
+        """
+        module_name = gen_uuid("udf")
```

Comment:
if possible and easy to do, maybe use operator id (maybe with worker id?) as a prefix?

Reply:
makes sense. let me try

Reply:
as module name is displayed to user on the frontend, as part of identifier for the line of code in log/print/exception trace, it's better to be concise and brief. Our engine maps operator, worker information on top of the udf code, those information can be found in other layers. It would be redundant to show worker and operator id in the module name.

Inspired by this comment, I have changed the module name to be a sequence of version number, instead of a uuid (which is very not user-friendly). Since we are using TempFS, which will ensure each worker has a unique temp directory, we do not need to worry about file name collision.